.github/workflows/cicd.yml

```diff
@@ -14,11 +14,7 @@ jobs:
     steps:
       - uses: actions/checkout@v3
       - uses: subosito/flutter-action@v2
```

Comment:
should we not still specify the version here, so the builds are more deterministic? in case they change the formatting algorithm

Reply:
I'm not sure, shouldn't we always make sure that the code is formatted according to the latest standard? Plus more work to maintain these numbers by having them static.

Reply:
Only a comment on this: I think in general it's much more work to update the code if the build fails due to a change to the analyzer than updating versions, while having other PRs in the pipeline. 
I once had this problem, which is only maintainable with static flutter versions:
- Having a PR (here PR1) which fixes the pipeline due to another error, which was mistakenly merged / flaky.
- Flutter upgrades the analyzer
- Need to update all the code to comply with the analyzer in PR2
- Cannot merge the PR1 or the PR2 as both fail due to analyzer / tree status, so need to merge them together in one PR although they not relate to each other.

Another use case is, if want to make a release and you can't merge the release PR as the build suddenly fails to to changes of the analyzer.

So I think the code only should match the analyzers latest changes, if a developer made sure in a PR that everything is working. This makes the workflow much more predictable. Don't get me wrong, I always want to make sure everything is up to date, but only if it have checked before if it really works.

A solution is to introduce a variable in GH Actions so one does not need to update all versions across the files.

Reply:
We could have so that it analyzes one static version and also the latest, but having the latest as a non-mandatory check.

Reply:
If that's possible to not let the pipeline fail, that's even better to then have some kind of warning. It's only a proposal as I don't contribute that often here :D

Reply:
Oooh, this was in the Flame repository, I though this was in the Audioplayers repo.
We haven't had any problems with the versions here that we haven't been able to quickly fix so far, so here I think we'll keep it like it is. :)

Reply:
Yes, but back then the version solving was manually: https://github.com/flame-engine/flame/pull/2532/files#diff-6727e33ccc9195d67f0786e1384f8f1cdaf4090c3e77547943105bd2b28c99d0

The problem will occur again sooner or later, like it did in audioplayers 😅

Reply:
Most of the time we had it as non-pinned versions, the commit you're pointing at is one of the few where we have done updates like that. It will be an issue here too, but imho it is easier to solve it directly when it occurs than to maintain the versions in the CI with the chance that they are forgotten. We will always have open PRs here that will have to rebase on the update anyhow, the queue is never empty.

Since our constraints allow for the latest flutter/sdk version that hasn't passed a major version, I think everything should scream in red (we have the same for the min-version, which is pinned, but that is updated much more seldom). :smile:

Reply:
And PR1 in your example can always be merged without a passing pipeline by an admin, so we are of course never fully locked out, but we haven't had to do that either for a very long time.

Reply:
Good that we had a look here, I can see that we actually have the min version pinned to the wrong version in the pipeline.